solr/core/src/java/org/apache/solr/servlet/SolrDispatchFilter.java

```diff
@@ -398,32 +397,12 @@ public CoreContainer getCores() {
   
   @Override
   public void destroy() {
-    if (closeOnDestroy) {
-      close();
-    }
-  }
-  
-  public void close() {
-    CoreContainer cc = cores;
-    cores = null;
     try {
-      if (metricManager != null) {
-        try {
-          metricManager.unregisterGauges(registryName, metricTag);
-        } catch (NullPointerException e) {
-          // okay
-        } catch (Exception e) {
-          log.warn("Exception closing FileCleaningTracker", e);
-        } finally {
-          metricManager = null;
-        }
-      }
-    } finally {
-      if (cc != null) {
-        httpClient = null;
```

Comment:
Nulling things can matter in rare circumstances in a perfect world. If you don't have a thread linger of 10+ seconds set for tests, or don't want unnecessary between test waits, the randomized testing framework will treat a TERMINATED thread as a leaked thread and wait for it or fail on it and that generally takes GC activity to clear out. Whether this is important here, don't know. Don't think it matters with the current test config and setup anyway though.

Reply:
I removed the metricManager field.  Maybe should do the same for httpClient -- always fetch it on-demand.  Fewer fields makes a class easier to reason about, especially when the lifecycle is governed by other fields!